torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -264,6 +264,19 @@ RegisterOperators reg({
             return 0;
           };
         }),
+    Operator(
+      prim::TupleIndex,
+      [](Node* node) {
+        return [](Stack& stack) {
+          int64_t index;
+          pop(stack, index);
+          auto tup = pop(stack).toTuple();
+          const auto & elems = tup->elements();
+          // index in-bounds check at compile time
+          stack.insert(stack.end(), elems.begin() + index, elems.begin() + index + 1);
```

Comment:
Isn't that just `stack.push_back(elems.at(index))`? I really think we should have run-time checks too, just to be safe.